feature: Add annotation support for enum configuration

flrossetto opened the pull request:
- Add new EnumConfig struct with annotation parsing capabilities
- Implement annotation processing for per-enum configuration overrides
- Support annotations like @marshal, @sql, @prefix, @nocase, @noprefix
- Add example annotation enum with comprehensive test coverage
- Update generator to respect annotation-based configuration
- Maintain backward compatibility with existing enum definitions

flrossetto committed:
feature: Add annotation support for enum configuration

- Add new EnumConfig struct with annotation parsing capabilities
- Implement annotation processing for per-enum configuration overrides
- Support annotations like @marshal, @sql, @prefix, @nocase, @noprefix
- Add example annotation enum with comprehensive test coverage
- Update generator to respect annotation-based configuration
- Maintain backward compatibility with existing enum definitions

abice commented:
Hey @flrossetto

I like the idea of adding per enum configuration, I've just bee too busy to give this a full review. I'll try to get to it over the holiday season! Thanks so much for spending time to improve the project.

abice commented on generator/enum_config.go:
```diff
@@ -0,0 +1,188 @@
+)
+
+// EnumConfigValue holds a configuration value with its validity flag.
+type EnumConfigValue struct {
```
Instead of using interface here, I'd prefer seeing this as a generic implementation.  I don't believe i'm supporting go versions older than 1.18 anymore, so I think it's ok to introduce this as 

```
EnumConfigValue[T enumValue] {
Value T
Valid bool
}
```

With the enumValue being the interface to constrain it to a string or bool.

Is there something I'm missing that would make that invalid?

flrossetto committed:
feat(generator): refactor EnumConfigValue to use generics for type safety

- Convert EnumConfigValue to generic struct with string or bool type constraint
- Update GetBool and GetString methods to use typed receivers, removing type assertions
- Modify EnumConfig fields to use typed EnumConfigValue instances
- Improves type safety and eliminates runtime type checking overhead

abice commented:
@flrossetto I don't seem to be able to run checks on your latest commit.  I have the "verified" validation on, so I'm not sure if that means that your last commit didn't have a verified signature, and that's why I can't kick off the checks, or if it's something else.  Do you mind getting that commit updated so I can run the checks on this PR?

flrossetto committed:
feat(generator): add MarshalText, UnmarshalText and AppendText methods to Null types

- Add MarshalText, UnmarshalText and AppendText methods to Null{{.enum.Name}} type
- Add MarshalText, UnmarshalText and AppendText methods to Null{{.enum.Name}}Str type
- These methods allow Null types to properly implement text marshalling interfaces
- Methods are only generated when .marshal flag is enabled
- Maintains backward compatibility with existing code

abice commented:
@flrossetto 

Looks like the commits still aren't signed, but the tests worked.

Also see that test coverage took a big hit on this one.  Mind getting that back up over the 90% mark?
